utils/serde/src/main/java/io/apicurio/registry/utils/serde/JsonSchemaKafkaSerializer.java

```diff
@@ -0,0 +1,188 @@
+/*
+ * Copyright 2020 Red Hat
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.apicurio.registry.utils.serde;
+
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.nio.ByteBuffer;
+import java.nio.charset.StandardCharsets;
+import java.util.Map;
+
+import javax.ws.rs.core.Response;
+
+import org.apache.kafka.common.errors.SerializationException;
+import org.apache.kafka.common.header.Headers;
+import org.apache.kafka.common.serialization.Serializer;
+
+import com.fasterxml.jackson.core.JsonGenerator;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.worldturner.medeia.api.StringSchemaSource;
+import com.worldturner.medeia.api.jackson.MedeiaJacksonApi;
+import com.worldturner.medeia.schema.validation.SchemaValidator;
+
+import io.apicurio.registry.client.RegistryService;
+import io.apicurio.registry.types.ArtifactType;
+import io.apicurio.registry.utils.serde.strategy.FindLatestIdStrategy;
+import io.apicurio.registry.utils.serde.strategy.GlobalIdStrategy;
+import io.apicurio.registry.utils.serde.util.Utils;
+
+/**
+ * An implementation of the Kafka Serializer for JSON Schema use-cases. This serializer assumes that the
+ * user's application needs to serialize a Java Bean to JSON data using Jackson. In addition to standard
+ * serialization of the bean, this implementation can also optionally validate it against a JSON schema.
+ * 
+ * @author eric.wittmann@gmail.com
+ */
+public class JsonSchemaKafkaSerializer<T>
+        extends AbstractKafkaStrategyAwareSerDe<SchemaValidator, JsonSchemaKafkaSerializer<T>>
+        implements Serializer<T> {
+    
+    public static final String REGISTRY_JSON_SCHEMA_SERIALIZER_VALIDATION_ENABLED = "apicurio.registry.serdes.json-schema.validation-enabled";
+
+    private static MedeiaJacksonApi api = new MedeiaJacksonApi();
+    private static ObjectMapper mapper = new ObjectMapper();
+    private static GlobalIdStrategy<SchemaValidator> latestVersionStrategy = new FindLatestIdStrategy<>();
```

Comment:
There is already an instance of GlobalIdStrategy in AbstractKafkaStrategyAwareSerDe ... which this class extends.

Reply:
Ah, OK, I see ... you need 2 ...